scripts/git-precommit-hook

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# NOTE: This file needs to be POSIX-shell compatible. No bash-isms, no zsh-isms, no fish-isms.
+
+if ! cargo fmt -- --help > /dev/null; then
+    printf '%b' "\033[33mwarning: cargo fmt not installed\033[0m\n" 1>&2
+    exit 0
+fi
+
+if ! which grep > /dev/null; then
+    printf '%b' "\033[33mwarning: grep not installed\033[0m\n" 1>&2
+    exit 0
+fi
+
+changed_files="`git diff-index --cached --name-only HEAD | grep \\.rs$`"
+
+if [ -z "$changed_files" ]; then
+    exit 0
+fi
+
+# nullbytes + xargs to pass filenames with spaces correctly to cargo fmt.
+if ! echo "$changed_files" | tr '\n' '\0' | xargs -0 cargo fmt -- --check --color=always; then
```

Comment:
- This should run `cargo +stable` just in case someone has a nightly override
- This should run `--all`
- Can we run the `make style` target to also check python code? Make is a development requirement anyway, and so is Python.